gateway/src/cluster/impl.rs

```diff
@@ -158,6 +163,7 @@ impl Cluster {
         }
 
         let shards = (from..=to)
+            .step_by(step as usize)
```

Comment:
Instead of casting here in all cases, the cast should be in the `ShardScheme::Bucket` variant